libs/designsystem/src/lib/components/radio/radio-group/radio-group.component.spec.ts

```diff
@@ -697,4 +704,540 @@ describe('RadioGroupComponent', () => {
+          let newSelectedIndex: number;
+
+          it('should update the bound ngModel when clicking a radio item', () => {
+            newSelectedIndex = defaultSelectedIndex + 1;
+            spectator.click(ionRadioElements[newSelectedIndex]);
+            expect(spectator.hostComponent.selected).toEqual(items[newSelectedIndex]);
+          });
+
+          it('should update the value of ion-radio-group when the bound ngModel is updated', async () => {
+            const newSelectedValue = items[defaultSelectedIndex + 1];
+            await setSelectedOnHostComponent(items[newSelectedValue]);
+            expect(ionRadioGroup.value).toEqual(items[newSelectedValue]);
+          });
+
+          it('should update the selected radio when the bound ngModel is updated', async () => {
+            newSelectedIndex = defaultSelectedIndex + 1;
+            await setSelectedOnHostComponent(items[newSelectedIndex]);
+            // Wait for radio checked attribute to be updated;
+            await TestHelper.whenTrue(() => radioChecked(newSelectedIndex));
+
+            expect(radioChecked(0)).toBeFalse();
+            expect(radioChecked(1)).toBeFalse();
+            expect(radioChecked(2)).toBeTrue();
+          });
+
+          it('should not emit change event when the bound ngModel is updated', async () => {
+            const onChangeSpy = spyOn(spectator.component.valueChange, 'emit');
+            await setSelectedOnHostComponent(items[2]);
+            expect(onChangeSpy).not.toHaveBeenCalled();
+          });
+        });
+      });
+
+      describe('error state when ngModel is required', () => {
+        beforeEach(async () => {
+          spectator = createHost(
+            `<kirby-radio-group [items]="items" [(ngModel)]="selected" required>
+            </kirby-radio-group>`,
+            {
+              hostProps: {
+                items: items,
+                selected: null,
+              },
+            }
+          );
+        });
+
+        describe('when ngModel is not null', () => {
+          beforeEach(async () => {
+            await setSelectedOnHostComponent(items[defaultSelectedIndex]);
+          });
+
+          describe('and ngModel is touched', async () => {
+            beforeEach(async () => {
+              spectator.component._onRadioBlur();
+              await TestHelper.waitForTimeout();
+              spectator.detectChanges();
+
+              expect(spectator.element.classList).toContain('ng-touched');
+            });
+
+            it('should not be in error state', async () => {
+              expect(spectator.element.classList).not.toContain('ng-invalid');
+            });
+          });
+
+          describe('and ngModul is untouched', async () => {
+            it('should not be in error state', async () => {
+              expect(spectator.element.classList).not.toContain('ng-touched');
+              expect(spectator.element.classList).not.toContain('ng-invalid');
```

Comment:
Consider expecting the rendering of the radios instead:
```suggestion
        //TODO loop radioElements, get icon for each and assert on rendered style:
         expect(radioIcon).toHaveComputedStyle({
        'border-width': '1px',
        'border-color': getColor('semi-dark'),
      });
```